### PR TITLE
[vtctld] migrate validator rpcs

### PR DESCRIPTION
**Pull request by ajm188**
## Description

This PR migrates the `Validate` and `Validate{Keyspace, Shard}` rpcs to the new VtctldServer.

<details><summary>Examples</summary>

```
❯ vtctldclient --server ":15999" Validate
Validation complete; no issues found.
❯ vtctldclient --server ":15999" ValidateKeyspace commerce
Validation of commerce complete; no issues found.
❯ vtctldclient --server ":15999" ValidateShard commerce/0
Validation of commerce/0 complete; no issues found.
❯ vtctldclient --server ":15999" ValidateShard commerce/0 -p
Validation of commerce/0 complete; no issues found.
❯ ps aux | grep vttablet | grep -v grep | shuf -n 1
amason           47526   0.0  0.2  5055136  75312 s000  S     6:46AM   0:00.37 vttablet -topo_implementation etcd2 -topo_global_server_address localhost:2379 -topo_global_root /vitess/global -log_dir /Users/amason/work/vitess/vtdataroot/tmp -log_queries_to_file /Users/amason/work/vitess/vtdataroot/tmp/vttablet_0000000101_querylog.txt -tablet-path zone1-0000000101 -tablet_hostname  -init_keyspace commerce -init_shard 0 -init_tablet_type replica -health_check_interval 5s -enable_semi_sync -enable_replication_reporter -backup_storage_implementation file -file_backup_storage_root /Users/amason/work/vitess/vtdataroot/backups -restore_from_backup -port 15101 -grpc_port 16101 -service_map grpc-queryservice,grpc-tabletmanager,grpc-updatestream -pid_file /Users/amason/work/vitess/vtdataroot/vt_0000000101/vttablet.pid -vtctld_addr http://SFO-M-AMASON02:15000/
❯ kill -9 47526
❯ vtctldclient --server ":15999" ValidateShard commerce/0 -p
Validation results:
- Ping(zone1-0000000101) failed: rpc error: code = DeadlineExceeded desc = latest balancer error: connection error: desc = "transport: Error while dialing dial tcp [::1]:16101: connect: connection refused" tablet hostname: localhost
E0923 06:48:01.531351   48260 main.go:42] shard commerce/0 had validation issues; see above for details
❯ vtctldclient --server ":15999" Validate -p
Validation results:

For commerce:

For commerce/0:
- Ping(zone1-0000000101) failed: rpc error: code = DeadlineExceeded desc = latest balancer error: connection error: desc = "transport: Error while dialing dial tcp [::1]:16101: connect: connection refused" tablet hostname: localhost
E0923 06:48:46.939330   48287 main.go:42] some issues were found during validation; see above for details
```

</details>

## Related Issue(s)

Closes https://github.com/vitessio/vitess/issues/8847
Closes https://github.com/vitessio/vitess/issues/8848

## Checklist
- [x] Should this PR be backported? - **No**
- [x] Tests were added or are not required
- [x] Documentation was added or is not required

### CODE REVIEW

**Comment by ajm188** [comment]
unit tests and cli entrypoints to come tomorrowwwww

File: go/vt/vtctl/grpcvtctldserver/server.go [hunk 1]
```diff
@@ -2169,6 +2170,368 @@ func (s *VtctldServer) UpdateCellsAlias(ctx context.Context, req *vtctldatapb.Up
...
 
+// Validate is part of the vtctlservicepb.VtctldServer interface.
+func (s *VtctldServer) Validate(ctx context.Context, req *vtctldatapb.ValidateRequest) (*vtctldatapb.ValidateResponse, error) {
+	span, ctx := trace.NewSpan(ctx, "VtctldServer.Validate")
+	defer span.Finish()
+
+	span.Annotate("ping_tablets", req.PingTablets)
+
+	resp := vtctldatapb.ValidateResponse{}
+	getKeyspacesCtx, getKeyspacesCancel := context.WithTimeout(ctx, *topo.RemoteOperationTimeout)
```
Comment (ajm188) [inline go/vt/vtctl/grpcvtctldserver/server.go:2181]: One change to the implementation is I bounded all the remote calls (both to the topo and to tablet tmc calls) with timeouts. If I didn't do this, I could kill a tablet, and the Validate call would hang until action_timeout, which defaults to 24 hours.

File: go/vt/vtctl/grpcvtctldserver/server.go [hunk 2]
```diff
@@ -2169,6 +2170,368 @@ func (s *VtctldServer) UpdateCellsAlias(ctx context.Context, req *vtctldatapb.Up
...
+
+	var (
+		m  sync.Mutex
+		wg sync.WaitGroup
+	)
+
+	wg.Add(1)
+	go func() {
+		defer wg.Done()
+		validateAllTablets := func(ctx context.Context, keyspaces []string) {
```
Comment (ajm188) [inline go/vt/vtctl/grpcvtctldserver/server.go:2198]: Another deviation from the original implementation: we used to make a call to `GetKeyspaces` in the outer function, and then the first thing this goroutine would do is ....... also make a call to `GetKeyspaces`, so, I just removed it and used the earlier result.

**Review by ajm188 — COMMENTED** [review]
just calling out deviations inline; this one was not a pure porting of behavior because I saw some things that could be improved

**Review by deepthi — APPROVED** [review]
LGTM

**Pull request merged**